src/roswire/ros2/msg.py

```diff
@@ -2,73 +2,74 @@
 __all__ = ("ROS2Field")
 
 import re
-from typing import Any, Dict, Optional
+from typing import Any, Dict, List, Optional
 
 import attr
 import dockerblade
 
 from ..common import Field
-from ..common.msg import MsgFormat, R_COMMENT
+from ..common.msg import Constant, MsgFormat, R_BLANK, R_COMMENT
+from ..exceptions import ParsingError
 
 
 @attr.s(frozen=True, str=False, slots=True, auto_attribs=True)
 class ROS2Field(Field):
     R_DEFAULT_VALUE = r"[^#]*"
-    R_FIELD = re.compile(f"^\s*({Field.R_TYPE})(\s+)({Field.R_NAME})"
-                         f"\s+({R_DEFAULT_VALUE})\s*"
-                         f"{R_COMMENT}$")
+    R_FIELD = re.compile(f"^\s*(?P<type>{Field.R_TYPE})"
+                         f"\s+(?P<name>{Field.R_NAME})(?:\s+)?"
+                         f"(?P<val>{R_DEFAULT_VALUE}){R_COMMENT}")
 
-    default_val: Optional[str]
+    val: Optional[str]
 
     @classmethod
-    def from_string(cls, packages: str, line: str) -> Optional["ROS2Field"]:
-        old_field = Field.from_string(packages, line)
+    def from_string(cls, package: str, line: str) -> Optional["ROS2Field"]:
         m_field = cls.R_FIELD.match(line)
-        if old_field and m_field:
-            def_val = m_field.group(4)
-
-            field: ROS2Field = ROS2Field(old_field.typ,
-                                         old_field.name,
-                                         def_val)
+        if m_field:
+            typ = m_field.group('type')
+            name = m_field.group('name')
+            typ = cls._resolve_type(package, typ)
+            val = m_field.group('val')
+            field = ROS2Field(typ, name, val if val and val != '' else None)
```

Comment:
`val` cannot be an empty string if it evaluates to `False`, so just `if val else` is equivalent.